### PR TITLE
Dispose IDisposables in RegexGenerator

### PR DESCRIPTION
**Pull request by omajid**
The StringWriter and IndentedTextWriter are IDisposable classes, so we should dispose them when we are done with them.

### CODE REVIEW

**Review by stephentoub — APPROVED** [review]
Thanks. This doesn't hurt anything, but fwiw it doesn't really help anything, either: these Dispose's are nops.

**Comment by omajid** [comment]
Agreed. I am doing some static analysis using some tools and a tool is flagging these uses. Would you prefer to leave this out?

**Comment by stephentoub** [comment]
> Agreed. I am doing some static analysis using some tools and a tool is flagging these uses. Would you prefer to leave this out?

No, it's fine. Like I said, it won't hurt :)

File: src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.cs [hunk 1]
```diff
@@ -140,8 +140,8 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                 }
 
                 // At this point we'll be emitting code.  Create a writer to hold it all.
-                var sw = new StringWriter();
-                IndentedTextWriter writer = new(sw);
+                using var sw = new StringWriter();
+                using IndentedTextWriter writer = new(sw);
```
Comment (gfoidl) [inline src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.cs:144]:
```suggestion
                using StringWriter sw = new();
                using IndentedTextWriter writer = new(sw);
```
So both have the same style?

**Pull request merged**